### PR TITLE
Added Iot credential usage to producer c samples

### PR DESCRIPTION
**Pull request by jdelapla**
*Issue #, if available:*

*Description of changes:*

- Add IoT credential provider usage to samples
- Refactor the samples to have a common header file for all samples to avoid duplication of #defines.

Testing:
- Ran the generate-iot script to verify the IoT thing is created and related IoT params
- Ran every sample by uncommenting the `IOT_CORE_ENABLE_CREDENTIALS` in the code to verify IoT credential provider works.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jdelapla** [comment]
Do not review yet, still checking some things.

**Review by hassanctech — APPROVED** [review]
lgtm

File: samples/KvsAudioOnlyStreamingSample.c [hunk 1]
```diff
@@ -110,21 +109,32 @@ INT32 main(INT32 argc, CHAR* argv[])
...
-    if (argc >= 5) {
-        if (!STRCMP(argv[2], AUDIO_CODEC_NAME_ALAW)) {
-            STRNCPY(audioCodec, AUDIO_CODEC_NAME_ALAW, STRLEN(AUDIO_CODEC_NAME_ALAW));
-        }
-    }
+#ifdef IOT_CORE_ENABLE_CREDENTIALS
+    PCHAR pIotCoreCredentialEndpoint, pIotCoreCert, pIotCorePrivateKey, pIotCoreRoleAlias, pIotCoreThingName;
+    CHK_ERR((pIotCoreCredentialEndpoint = getenv(IOT_CORE_CREDENTIAL_ENDPOINT)) != NULL, STATUS_INVALID_OPERATION,
+            "AWS_IOT_CORE_CREDENTIAL_ENDPOINT must be set");
+    CHK_ERR((pIotCoreCert = getenv(IOT_CORE_CERT)) != NULL, STATUS_INVALID_OPERATION, "AWS_IOT_CORE_CERT must be set");
```
Comment (sirknightj) [inline samples/KvsAudioOnlyStreamingSample.c:116]: Prefer GETENV macro

Comment (sirknightj) [reply samples/KvsAudioOnlyStreamingSample.c:116]: https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/57637ea593f4b43c509413a44d993ed08d7f2616/src/common/include/com/amazonaws/kinesis/video/common/CommonDefs.h#L852

**Review by sirknightj — APPROVED** [review]

**Pull request merged**